### DISCUSSION TITLE
CKAN API reference for custom set of extensions

### DISCUSSION THREAD

**Discussion by tomeksabala** [ckan/ckan#6174] [Ideas]
It'd be great to have an easy way to generate custom "CKAN API reference" documenting all custom actions added by various extensions for my custom CKAN stack.
https://docs.ckan.org/en/2.9/api/index.html#action-api-reference is a great resource. I would like to generate my own version which would include docstrings of all my extensions implementing IActions interface.

**Comment by wardi** [comment]
+1 and even better would be generating the docs in swagger format. For our particular site being able to generate different language API docs from the same source would be ideal

**Reply by jqnatividad** [reply]
Has the team looked at [flask-restplus](https://flask-restplus.readthedocs.io/en/stable/index.html) or [flask-restx](https://flask-restx.readthedocs.io/en/latest/)? It automatically generates swagger documentation by using decorators. It even allows you to export your API as a Postman collection.
